server/utils/bot_poster.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package utils
+
+import (
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/config"
+)
+
+type BotPoster interface {
+	PostDirect(userID, message, postType string) error
+	PostEphemeral(userID, channelId, message string)
```

Comment:
Is there a use case where we post a publicly in a channel? We can just add it when we need it.